DOCS.md

```diff
@@ -0,0 +1,367 @@
+# API Documentation
+
+## `compile` Function
+
+The `compile` function is the core functionality of eval-magic that parses, transforms, and prepares JavaScript code for execution with enhanced features like operator overloading, custom imports, and export handling.
+
+### Signature
+
+```typescript
+function compile(
+  code: string,
+  globals?: Record<string, any>,
+  options?: RunOptions
+): CompiledCode
+```
+
+### Parameters
+
+#### `code: string`
+The JavaScript code to compile. This can include:
+- Standard JavaScript syntax
+- Export statements (when `options.returns` is "exports")
+- Import statements (when `options.importFunction` is provided)
+- Operator overloading expressions (when `options.operatorOverloading` is enabled)
+
+#### `globals?: Record<string, any>` (optional)
+An object containing variables and functions to make available in the code's execution scope. Only identifiers that are actually referenced in the code will be included in the final scope for security and performance.
+
+**Default:** `{}`
+
+#### `options?: RunOptions` (optional)
+Configuration options for parsing, transforming, and executing the code.
+
+**Default:** 
+```typescript
+{
+  parseOptions: { ecmaVersion: "latest" },
+  codegenOptions: {},
+  returns: "exports",
+  operatorOverloading: true,
+  customVisitors: {},
+  isAsync: false
+}
+```
+
+**Properties:**
+
+- **`parseOptions?: Partial<acorn.Options>`**
+  
+  Options passed to the Acorn parser. 
+  
+  **Default:**
+  ```typescript
+  {
+    ecmaVersion: "latest",
+    sourceType: "module", // when options.returns === "exports"
+    allowReturnOutsideFunction: true // when options.returns === "return"
+  }
+  ```
+
+- **`codegenOptions?: escodegen.GenerateOptions`**
+  
+  Options passed to Escodegen for code generation from the transformed AST.
+
+- **`returns?: "exports" | "return"`**
+  
+  Defines how the code should handle return values:
+  - `"exports"` (default): Use export statements to return values, enables modular code patterns
+  - `"return"`: Use standard JavaScript return statements
+  
+- **`operatorOverloading?: boolean`**
+  
+  Enables Python-like operator overloading via magic methods using the `Py` symbols.
+  
+  **Default:** `true`
+
+- **`customVisitors?: Record<string, VisitorFn>`**
+  
+  Custom AST visitor functions for additional transformations during the compilation process.
+  
+  **Note:** This property is not currently defined in the TypeScript interface but is used in the implementation.
+
+- **`importFunction?: (source: string) => Object | Promise<Object>`**
+  
+  Custom import handler for resolving module imports. Can be synchronous or asynchronous.
+  
+  **Note:** If an async function is provided, `options.isAsync` will automatically be set to `true`.
+
+- **`isAsync?: boolean`**
+  
+  Wraps the generated code in an async function, enabling top-level await usage.
+  
```

Comment:
Mention that even with explicit `isAsync: false` the function returned by compile will still return a promise (will be async) if:
- any `await` keyword appears in the code.
- your code have import statement, and the passed importFunction is async.

Add TODO comment to check for await keywords, they should affect the async behavior only if they are used on the root level, because if they are used internally in an async function defined in the code, they shouldn't change the compile function behavior